src/main/java/build/buildfarm/worker/Executor.java

```diff
@@ -200,7 +201,12 @@ private long executePolled(
           arguments.addAll(transformWrapper(policy.getWrapper()));
         }
       }
-      arguments.addAll(command.getArgumentsList());
+
+      // Make sure that the executable path is absolute, otherwise processbuilder fails on windows
+      Iterator<String> argumentItr = command.getArgumentsList().iterator();
+      String exe = argumentItr.next(); // Get first element, this is the executable
+      arguments.add(workingDirectory.resolve(exe).toAbsolutePath().normalize().toString());
+      argumentItr.forEachRemaining(arg -> arguments.add(arg));
```

Comment:
Please change this to only be effective for absolute path on windows.

Also, it is a violation of iterator usage to call next() without calling hasNext() (and reacting to it)

Reply:
windows only makes sense